v2/invoke_test.go

```diff
@@ -68,6 +72,47 @@ func TestInvokeSuccess(t *testing.T) {
 		t.Errorf("slept %d times, should not have slept since the call succeeded", int(sp))
 	}
 }
+func TestInvokeAPIError(t *testing.T) {
+	br := &errdetails.BadRequest{}
+	br_violation := &errdetails.BadRequest_FieldViolation{
+		Field:       "field",
+		Description: "desc",
+	}
+	br.FieldViolations = append(br.FieldViolations, br_violation)
+	br_stat, _ := status.New(codes.InvalidArgument, "bad request").WithDetails(br)
+	br_err := br_stat.Err()
+	br_msg := ErrDetails{}
+	br_msg.BadRequest = br
+	_ = APIError{
+		err:     br_err,
+		status:  br_stat,
+		details: br_msg,
+	}
+	apierr, _ := FromError(br_err)
+	apiCall := func(context.Context, CallSettings) error { return br_err }
+	var sp recordSleeper
+	err := invoke(context.Background(), apiCall, CallSettings{}, sp.sleep)
+	if err != apierr {
```

Comment:
As we talked about, try casting this to an `*APIError` then comparing the fields. OR use `github.com/google/go-cmp` to compare them. This is an appropriate dependency to add.

Example usage: https://github.com/googleapis/gapic-generator-go/blob/master/showcase/echo_test.go#L349-L351